l1-contracts/contracts/state-transition/chain-deps/facets/Mailbox.sol

```diff
@@ -226,6 +227,10 @@ contract MailboxFacet is ZkSyncStateTransitionBase, IMailbox {
     function _requestL2TransactionSender(
         BridgehubL2TransactionRequest memory _request
     ) internal nonReentrant returns (bytes32 canonicalTxHash) {
+        // Check that the transaction is allowed by the filterer (if the filterer is set).
+        if (s.transactionFilterer != address(0)) {
+            require(ITransactionFilterer(s.transactionFilterer).isTransactionAllowed(_request), "tf");
```

Comment:
Why do you need to submit the whole Transaction request for a check? This is gas-consuming, please only submit fields that makes sense to check (from?)

Reply:
My initial thought was to give as much flexibility as possible to the partners implementing the filters, but your point is valid so  I reduced it to `address sender, address contractL2, uint256 mintValue, uint256 l2Value`

Reply:
By the Stas comments, you may consider adding calldata `selector` this list

Reply:
done

Reply:
I meant selector (first 4 bytes of the calldata), not the whole calldata. This is to minimize the cost too

@StanislavBreadless wdyt?

Reply:
how is the selector going to be useful here? I thought that it would be more important to have the actual values of the calldata

Reply:
We talked with @vladbochok and it makes sense to pass the entire calldata